### PR TITLE
Book list

### PR DESCRIPTION
**Pull request by lesang458**

### CODE REVIEW

File: assignment-project/src/app/core/http/book.service.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { share, map} from 'rxjs/operators';
+
+import { Book } from '../../shared/models/book.model';
+import { PaginatedResult } from 'src/app/shared/models/pagination.model';
+import { environment } from 'src/environments/environment';
+import { Category } from 'src/app/shared/models/category.model';
+
+@Injectable({providedIn: 'root'})
+export class BookService {
+  private baseUrl = environment.apiUrl + 'books/';
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/book.service.ts:13]: Use template literal.

File: assignment-project/src/app/core/http/book.service.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      params = params.append('pageSize', itemsPerPage);
+    }
+
+    if (selectedCategories.length > 0) {
+      const categoriesIds = selectedCategories.reduce((accumulator, currentValue, currentIndex) => {
+        if (currentIndex === selectedCategories.length - 1) {
+          return accumulator + currentValue.id;
+        }
+        return accumulator + currentValue.id + ',';
+      }, '');
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/book.service.ts:33]: Use template literals.

File: assignment-project/src/app/core/http/book.service.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    if (page && itemsPerPage) {
+      params = params.append('bookName', bookName);
+      params = params.append('authorId', authorId);
+      params = params.append('pageNumber',  page);
+      params = params.append('pageSize', itemsPerPage);
+    }
+
+    if (selectedCategories.length > 0) {
+      const categoriesIds = selectedCategories.reduce((accumulator, currentValue, currentIndex) => {
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/book.service.ts:28]: if(selectedCategories.length)

File: assignment-project/src/app/core/http/category.service.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Observable } from 'rxjs/internal/Observable';
+
+import { Category } from 'src/app/shared/models/category.model';
+import { environment } from 'src/environments/environment';
+
+@Injectable({providedIn: 'root'})
+export class CategoryService {
+  private baseUrl = environment.apiUrl + 'categories/';
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/category.service.ts:10]: same here.

File: assignment-project/src/app/core/http/category.service.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      const categoriesIds = selectedCategories.reduce((accumulator, currentValue, currentIndex) => {
+        if (currentIndex === selectedCategories.length - 1) {
+          return accumulator + currentValue.id;
+        }
+        return accumulator + currentValue.id + ',';
+      }, '');
+      params = params.append('selectedCategories', categoriesIds);
+    }
+    return this.http.get<Category[]>(
+      this.baseUrl + 'search',
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/http/category.service.ts:25]: same here.

File: assignment-project/src/app/core/interceptors/token.interceptor.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class TokenInterceptor implements HttpInterceptor {
+
+  constructor() {}
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    const token = 'Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxIiwidW5pcXVlX25hbWUiOiJtYW5hZ2VyMSIsInJvbGUiOiJNQU5BR0VSIiwibmJmIjoxNTk1NDg4MDQxLCJleHAiOjE1OTU5MjAwNDEsImlhdCI6MTU5NTQ4ODA0MX0.DNJ5O2C5xfoF99H0rR68-9_4ckLazHuSZ-wk-CyuAeyaMI-7ichJA8ympO87jKCYSkV_9Y01QFmhgiGUi7Fx4Q';
+    const modifiedReq = request.clone({
```
Comment (tranmanhlinh) [inline assignment-project/src/app/core/interceptors/token.interceptor.ts:18]: Store token in the local storage.

File: assignment-project/src/app/core/http/book.service.ts (hunk 1 above)
Comment (lesang458) [reply assignment-project/src/app/core/http/book.service.ts]: done

File: assignment-project/src/app/core/http/book.service.ts (hunk 2 above)
Comment (lesang458) [reply assignment-project/src/app/core/http/book.service.ts]: done

File: assignment-project/src/app/core/http/category.service.ts (hunk 4 above)
Comment (lesang458) [reply assignment-project/src/app/core/http/category.service.ts]: done

File: assignment-project/src/app/core/http/category.service.ts (hunk 5 above)
Comment (lesang458) [reply assignment-project/src/app/core/http/category.service.ts]: done

File: assignment-project/src/app/core/interceptors/token.interceptor.ts (hunk 6 above)
Comment (lesang458) [reply assignment-project/src/app/core/interceptors/token.interceptor.ts:18]: done

**Pull request merged**